new page for adding a property manager

DaveTrost opened the pull request:
### What issue is this solving?
closes #480

At this point, there is limited functionality because backend routes don't support adding a user with an explicit role. Only a few lines of changes are needed once the backend functionality is complete. It may make sense to hold off on this merge until the backend PR is merged and then issue 480 can be closed in one go.

### Any helpful knowledge/context for the reviewer?
Is a `npm install` necessary? no
Any special requirements to test? Go to Add -> Property Manager

### Please make sure you've attempted to meet the following coding standards
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code
- [x] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

If you're having trouble meeting this criteria, feel free to reach out on Slack for help!

ccbass commented on src/views/addManager/addManager.js:
```diff
@@ -0,0 +1,280 @@
+  lastName: Yup.string()
+    .max(255, "Must be shorter than 255 Characters")
+    .required("Must enter a Last Name"),
+  phone: Yup.string()
```
Is there a reason the validation does length 5 - 20?  Accounting for non-standard/international #?

ccbass commented on src/views/addManager/addManager.js:
```diff
@@ -0,0 +1,280 @@
+  lastName: Yup.string()
+    .max(255, "Must be shorter than 255 Characters")
+    .required("Must enter a Last Name"),
+  phone: Yup.string()
+    .min(
+      5,
+      "*Number must contain at least 5 digits to be a valid phone/text number"
+    )
+    .max(20, "*Numbers can't be longer than 20 digits")
+    .required("*a valid phone number is required"),
```
Is there a reason the validation does length 5 - 20? Accounting for non-standard/international #?

DaveTrost replied:
The validation length was copied from another component in the app. It looks like there is some inconsistency across different components. Phone minimum lengths are 5 or 7 and the maximum lengths are 20 or 25. Different components use different combinations. 

Maybe it is a good idea to open an issue on this - something like having a common file in which to declare properties for use in the call to `Yup.object().shape()`?

aedwardg replied:
@ccbass @DaveTrost 
There is also an inconsistency between the frontend and backend on this, so maybe this is something we should bring up and resolve in the Wednesday meeting.

I think Nick made a pretty compelling argument for increasing the maximum length to 30 in [backend PR 183](https://github.com/codeforpdx/dwellinglybackend/pull/183). 

Given that any non-numeric characters are still counted toward the max and min lengths, and given the talk last meeting about this software potentially being used by other organizations down the line, I think it's probably better to future-proof it now by having a buffer between a standard number's characters and the field limits.

ccbass commented on src/views/addManager/addManager.js:
```diff
@@ -0,0 +1,280 @@
+    setPropertySearchResults(choices);
+  }, [propertySearchText, propertyOptions]);
+
+  const getProperties = () => {
+    axios.get("/api/properties", makeAuthHeaders(context))
+      .then(({ data }) => {
+        let properties = data.properties && data.properties.length > 0
+          ? data.properties.map(property => {
+            return {
+              key: property.id,
+              description: `${property.name}, ${property.address}`
+            };
+          })
+          : data.properties;
+        setPropertyOptions(properties);
+        setPropertySearchResults(properties);
+        setShowAddProperty(false);
+      });
+  };
```
Did you want to provide any error handling here?  (.catch/response==200, etc.)

DaveTrost replied:
@ccbass - I think Hugh may have addressed this concern in the latest commits to the PR. Might be worth a re-review.

DaveTrost replied:
I added some error checking here. Thanks for the heads-up @ccbass !

DaveTrost commented on src/views/addManager/addManager.js:
```diff
@@ -73,12 +73,23 @@ export const AddManager = () => {
       confirmPassword: 'changeAt1stLogin!',
-      // role: RoleEnum.PROPERTY_MANAGER,
-      // propertyIDs: propertySelection.map(p => p.key),
+      role: "PROPERTY_MANAGER",
```
We should be able to use the global enum instead of hard-coding the role here. Something like: `RoleEnum.PROPERTY_MANAGER`

DaveTrost commented on src/views/addManager/addManager.js:
```diff
@@ -81,7 +81,7 @@ export const AddManager = () => {
 
     axios
-      .post(`/api/user/add`, body, makeAuthHeaders(context))
+      .post(`/api/user/invite`, body, makeAuthHeaders(context))
```
The `/user/invite` endpoint has functionality to email the invited user. This is important information for the admin user to know, and it should be included in the Toast message (see line 93)

DaveTrost replied:
Nice touch by @htharker42: `Toast('Property Manager Created Successfully, an invite email has been sent.', "success");`

DaveTrost reviewed (commented):
Looking good @htharker42. I had a couple of ideas and made notes in the code - hope you find them helpful

Chris-Boe commented on src/views/Manager/index.js:
```diff
@@ -45,7 +45,7 @@ const getManager = (context, managerId, storeInState) => {
 const Manager = () => {
   const { pathname } = useLocation();
-  const id = pathname.match(/\d/)[0];
+  const id = pathname.match(/\d+/)[0];
```
For getting the id from the path, you can use the React hook: useParams()

htharker42 committed:
removed errant logging commands

htharker42 committed:
removed more errant logging

htharker42 committed:
connected processes to backend, chained api calls to connect user to properties, fixed error with regex on managers table which was truncating the user id to create the property manager edit page

DaveTrost committed:
new page for adding a property manager

limited functionality because backend routes don't support adding a user with an explicit role (PROPERTY_MANAGER).
Code heavily influenced by addTenant page
Add link from manage/managers page to this add/manager page

htharker42 committed:
leveraged user/invite rather than ancilliary endpoint I created

DaveTrost committed:
cleanup and error checking

use predefined variables for constant values
replace useLocaiton with useParams for getting id from the url

htharker42 committed:
added message to inform user email was dispatched"

Chris-Boe reviewed (approved):
This looks and works great!
